CMakePresets.json

```diff
@@ -0,0 +1,97 @@
+            "name": "common",
+            "hidden": true,
+            "generator": "Ninja",
+            "binaryDir": "build/${presetName}",
+            "cacheVariables": {
+                "CMAKE_CXX_STANDARD": "20",
+                "CMAKE_CXX_COMPILER_LAUNCHER": "ccache",
+                "CMAKE_EXPORT_COMPILE_COMMANDS": "ON",
+                "ACTS_ENABLE_LOG_FAILURE_THRESHOLD": "OFF",
+                "ACTS_BUILD_FATRAS": "ON",
+                "ACTS_BUILD_ODD": "ON",
+                "ACTS_BUILD_UNITTESTS": "ON",
+                "ACTS_BUILD_INTEGRATIONTESTS": "ON",
+                "ACTS_BUILD_PLUGIN_LEGACY": "ON",
+                "ACTS_BUILD_EXAMPLES_DD4HEP": "ON",
+                "ACTS_BUILD_EXAMPLES_PYTHON_BINDINGS": "ON",
+                "ACTS_BUILD_EXAMPLES_UNITTESTS": "ON"
+            }
+        },
+        {
+            "name": "dev",
+            "displayName": "Development",
+            "inherits": "common",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+                "CMAKE_EXPORT_COMPILE_COMMANDS": "ON",
+                "ACTS_FORCE_ASSERTIONS": "ON"
+            }
+        },
+        {
+            "name": "perf",
+            "displayName": "Performance",
+            "inherits": "common",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release",
+                "CMAKE_CXX_FLAGS_RELEASE": "-O3 -DNDEBUG -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer",
+                "ACTS_BUILD_BENCHMARKS": "ON"
+            }
+        },
+        {
+            "name": "ci-common",
+            "hidden": true,
+            "inherits": "common",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release",
+                "CMAKE_CXX_FLAGS_RELEASE": "-Werror",
+                "ACTS_FORCE_ASSERTIONS": "ON",
+                "ACTS_ENABLE_LOG_FAILURE_THRESHOLD": "ON",
+                "ACTS_BUILD_BENCHMARKS": "ON",
+                "ACTS_BUILD_NONCOMPILE_TESTS": "ON",
+                "ACTS_BUILD_ANALYSIS_APPS": "ON",
+                "ACTS_BUILD_ALIGNMENT": "ON",
+                "ACTS_BUILD_FATRAS_GEANT4": "ON",
+                "ACTS_BUILD_PLUGIN_ACTSVG": "ON",
+                "ACTS_BUILD_EXAMPLES_EDM4HEP": "ON",
+                "ACTS_BUILD_EXAMPLES_GEANT4": "ON",
+                "ACTS_BUILD_EXAMPLES_HEPMC3": "ON",
+                "ACTS_BUILD_EXAMPLES_PYTHIA8": "ON"
+            }
+        },
+        {
+            "name": "github-ci",
+            "displayName": "GitHub-CI",
+            "inherits": "ci-common",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
+            }
+        },
+        {
+            "name": "gitlab-ci",
+            "displayName": "GitLab-CI",
+            "inherits": "ci-common",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+                "ACTS_BUILD_PLUGIN_FPEMON": "ON"
+            }
+        }
+    ],
+    "buildPresets": [
```

Comment:
Sorry for my ignorance but what do these do?

Reply:
I think we can drop these. I believe these are presets for the build command `cmake --build --preset full-chain` or something like that. So you could collect targets and build them together. I don't think we really need this at this point